x/auth/ante/ante_test.go

```diff
@@ -481,7 +471,7 @@ func TestAnteHandlerBadSignBytes(t *testing.T) {
 	// test wrong signer if public key exist
 	privs, accnums, seqs = []crypto.PrivKey{priv2}, []uint64{0}, []uint64{1}
 	tx = types.NewTestTx(ctx, msgs, privs, accnums, seqs, fee)
-	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeUnauthorized)
+	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeInvalidPubKey)
```

Comment:
Is the different error key returned part of the decorator change?

Reply:
Yes, this is because the order of antehandler logic has changed. With AnteDecorator flow, we try to set PubKeys in account first before validating any of the signatures.

If the provided Pubkey in signature does not match the address, we error in `SetPubKeyDecorator` with an Invalid PubKey. This happens before any signature validation. Though we can still get `Unauthorized` errors if anything fails in the SignatureVerificationDecorator